### ISSUE TITLE
Add support for Scaleway in Kops

### ISSUE THREAD

**Issue by Mia-Cross** [kubernetes/kops#13954]
/kind feature

We at Scaleway would like to add support in Kops for our Kapsule product so that our users are able to deploy their cluster easily with Kops.

**Comment by olemarkus** [comment]
Isn't Kapsule a managed k8s cluster? kOps is a solution for easily managing the control plane itself. It would rather be built directly upon the compute offering.

**Comment by hakman** [comment]
If you would like to work with us to add support via the compute offering, it would be greatly appreciated.
To get an idea, here is how the Hetzner Cloud support was added https://github.com/kubernetes/kops/pull/13572.

**Comment by Mia-Cross** [comment]
My phrasing was off, sorry, I reworded the issue. Thank you @hakman for the link, it will be useful !

**Comment by hakman** [comment]
@Mia-Cross @remyleone is there any plan to continue supporting kOps by Scaleway?

**Comment by remyleone** [comment]
Yes is there improvements that would be required on your side for the latest kops development?

**Comment by hakman** [comment]
> Yes is there improvements that would be required on your side for the latest kops development?

I think there are a few major topics:
* testing - how can we get some credits to do minimal testing to not break things
* security - review node joining an identity
* bump CCM and CSI
* figure out if anything needs to be updated/improved

any thoughts?

**Comment by remyleone** [comment]
For the credit, I would suggest first to create an organisation on https://www.scaleway.com with an email address that is well integrated within the Kops/CNCF CI ecosystem. I'm not familiar enough but once you get this account I can look around to provide you enough voucher for testing.
As for the CCM/CSI is there any way to have this job performed by a dependabot or something similar?

**Comment by hakman** [comment]
> For the credit, I would suggest first to create an organisation on https://www.scaleway.com/ with an email address that is well integrated within the Kops/CNCF CI ecosystem. I'm not familiar enough but once you get this account I can look around to provide you enough voucher for testing.

ACK, will check with @ameukam and see how we can move this forward.

> As for the CCM/CSI is there any way to have this job performed by a dependabot or something similar?

Not really, for now there are still debates around what k8s org can use.
The main issue is not the actual upgrade but, without testing it's pretty risky, no way to know if anything will be silently broken.

**Comment by ameukam** [comment]
> For the credit, I would suggest first to create an organisation on [scaleway.com](https://www.scaleway.com) with an email address that is well integrated within the Kops/CNCF CI ecosystem. I'm not familiar enough but once you get this account I can look around to provide you enough voucher for testing. As for the CCM/CSI is there any way to have this job performed by a dependabot or something similar?

The general process would for the provider to reach to CNCF to establish a proper partnership. CNCF represents the legal entity of the project. All donations/agreements goes through this process.

**Comment by remyleone** [comment]
We are already a CNCF member. I'm not sure who to contact and what the process is.

**Comment by ameukam** [comment]
> We are already a CNCF member. I'm not sure who to contact and what the process is.

You can reach to `Jeffrey Sica <jeefy@linuxfoundation.org>`.

**Comment by remyleone** [comment]
I contacted @jeefy by email today thanks :)

**Comment by hakman** [comment]
Thank you @remyleone! 🙂
